pypeit/core/coadd.py

```diff
@@ -762,11 +762,10 @@ def sn_weights(waves, fluxes, ivars, masks, sn_smooth_npix, const_weights=False,
         # Adjust the arrays to be relative
         refscale = (sn_val[:, ref_spec] > 0) / (sn_val[:, ref_spec] + (sn_val[:, ref_spec] == 0))
         for iexp in range(nstack):
-            if iexp != ref_spec:
-                # Compute the relative (S/N)^2 and update the mask
-                sn2[iexp] /= sn2[ref_spec]
-                mask_stack[:, iexp] *= (mask_stack[:, ref_spec]) | (sn_val[:, ref_spec] != 0)
-                sn_val[:, iexp] *= refscale
+            # Compute the relative (S/N)^2 and update the mask
+            sn2[iexp] /= sn2[ref_spec]
+            mask_stack[:, iexp] *= (mask_stack[:, ref_spec]) | (sn_val[:, ref_spec] != 0)
```

Comment:
Formally should this not be a boolean operation rather than *=? I'm sure this does the same thing but is less clear since mask_stack is a boolean array.